src/python_framework_v02/troute/nhd_io.py

```diff
@@ -218,6 +218,70 @@ def drop_all_coords(ds):
     return ds.reset_coords(drop=True)
 
 
+def write_q_to_wrf_hydro(flowveldepth, chrtout_files, qts_subdivisions):
+    """
+    Write t-route simulated flows to WRF-Hydro CHRTOUT files. 
+    
+    Arguments:
+        flowveldepth (pandas Data Frame): t-route simulated flow, velocity and depth
+        chrtout_files (list): chrtout filepaths
+        qts_subdivisions (int): number of t-route timesteps per WRF-hydro timesteps 
+    """
+        
+    # open all CHRTOUT files as a single xarray dataset
+    with xr.open_mfdataset(
+        chrtout_files,
+        combine="by_coords"
+    ) as chrtout:
+
+        # !!NOTE: If break_at_waterbodies == True, segment feature_ids coincident with water bodies do
+        # not show up in the flowveldepth dataframe. Re-indexing inserts these missing feature_ids and
+        # populates columns with NaN values. 
+        flowveldepth_reindex = flowveldepth.reindex(chrtout.feature_id.values)
+
+        # unpack, subset, and transpose t-route flow data
+        qtrt = flowveldepth_reindex.loc[:,::3]
+            .to_numpy()
+            .astype("float32")
+        qtrt = qtrt[:,::qts_subdivisions]
+        qtrt = np.transpose(qtrt)
+
+        # construct DataArray for t-route flows, dims, coords, and attrs consistent with CHRTOUT
+        qtrt_DataArray = xr.DataArray(
+            data = da.from_array(qtrt),
+            dims = ["time","feature_id"],
+            coords = dict(
+                time = chrtout.time.values,
+                feature_id = chrtout.feature_id.values
+            ),
+            attrs = dict(
+                description = "River Flow, t-route",
+                units = "m3 s-1",
+            ),
+        )
+
+        # add t-route DataArray to CHRTOUT dataset
+        chrtout["streamflow_troute"] = qtrt_DataArray
+
+        # group by time
+        grp_object = chrtout.groupby("time")
```

Comment:
If time is already one of the dimensions, do we really need a groupby here?

Reply:
Good question. I suspect that when we subject this write procedure to a full CONUS test dataset (only available on the Cheyenne HPC, at the moment) we will need to make a number of adjustments to address performance. I'm going to create an issue staging that HPC test (and associated refactor) and include the note that we should confirm the need (or not) of this `groupby`.